### PR TITLE
send http responses in header rather than body

### PR DESCRIPTION
**Pull request by nickcurie**
## Notes
* Moved logging to use opencost's version of logs
* Moved the deprecated `ioutil` calls to `os` / `io`
* Removed unnecessary parameters for `Turndown Endpoints` struct 
* Utilize opencost's `protocol` package for api responses
## Testing
* Tested endpoints in unit tests

### CODE REVIEW

**Review by avrodrigues5 — COMMENTED** [review]
Based on the code, it looks good that you have moved the error code from the response body to the response JSON. 

But based on the original ticket, https://kubecost.atlassian.net/issues/ENG-751?filter=10157, there's a 500 error code. Are you able to repro that and see why that is happening? The ticket says cannot create a Turndown schedule.

File: pkg/turndown/turndownendpoints.go [hunk 1]
```diff
@@ -62,26 +49,29 @@ func (te *TurndownEndpoints) HandleStartSchedule(w http.ResponseWriter, r *http.
 
 	if r.Method == http.MethodGet {
 		schedule := te.scheduler.GetSchedule()
-		if schedule == nil {
-			w.Write(wrapData(struct{}{}, nil))
+		log.Infof("get request")
```
Comment (nik-kc) [inline pkg/turndown/turndownendpoints.go]: I'm guessing this line was accidentally committed?

File: pkg/turndown/turndownendpoints_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	mockTurndownProvider := provider.NewMockTurndownProvider(mockProvider)
+	mockStrategy := strategy.NewStandardTurndownStrategy(nil, mockTurndownProvider)
+	mockManager := NewKubernetesTurndownManager(nil, mockTurndownProvider, mockStrategy, node)
+
+	mockScheduler := NewTurndownScheduler(mockManager, mockScheduleStore)
+
+	te := NewMockTurndownEndpoints(mockClient, mockScheduler, mockManager)
+
+	//invalid body
+	body := strings.NewReader("Hi")
```
Comment (nik-kc) [inline pkg/turndown/turndownendpoints_test.go:76]: I'm glad to see that I'm not the only one who uses greetings as test strings.

**Review by nik-kc — APPROVED** [review]
Looks great! One tiny tiny potential change, but otherwise, I approve.

**Comment by nik-kc** [comment]
> Based on the code, it looks good that you have moved the error code from the response body to the response JSON.
> 
> But based on the original ticket, https://kubecost.atlassian.net/issues/ENG-751?filter=10157, there's a 500 error code. Are you able to repro that and see why that is happening? The ticket says cannot create a Turndown schedule.

Good catch, Alan, I didn't notice this. I echo this question.

File: pkg/turndown/turndownendpoints.go (hunk 1 above)
Comment (nickcurie) [reply pkg/turndown/turndownendpoints.go]: Yes it was :) thanks

**Comment by nickcurie** [comment]
Yeah if you see my comment here: https://kubecost.atlassian.net/browse/ENG-751?focusedCommentId=124848 the reason why we got the creation failure was due to the k8s api being unresponsive. I recall when I first started working on the ticket we needed to restart the pod because of this. The original issue was not reproducible after that point, so I just proceeded with modernizing parts of this repo since it was written 4 years ago and hasn't really been touched since

**Comment by nik-kc** [comment]
> Yeah if you see my comment here: https://kubecost.atlassian.net/browse/ENG-751?focusedCommentId=124848 the reason why we got the creation failure was due to the k8s api being unresponsive. I recall when I first started working on the ticket we needed to restart the pod because of this. The original issue was not reproducible after that point, so I just proceeded with modernizing parts of this repo since it was written 4 years ago and hasn't really been touched since

I'm happy with that, I think this is good to merge. What say you, @avrodrigues5?

**Comment by avrodrigues5** [comment]
> > Yeah if you see my comment here: https://kubecost.atlassian.net/browse/ENG-751?focusedCommentId=124848 the reason why we got the creation failure was due to the k8s api being unresponsive. I recall when I first started working on the ticket we needed to restart the pod because of this. The original issue was not reproducible after that point, so I just proceeded with modernizing parts of this repo since it was written 4 years ago and hasn't really been touched since
> 
> I'm happy with that, I think this is good to merge. What say you, @avrodrigues5?

Yup makes sense to me too. Thanks @nickcurie

**Pull request merged**